x-pack/test_serverless/api_integration/test_suites/observability/apm_api_integration/traces/critical_path.ts

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import expect from 'expect';
+import { apm, ApmFields, SynthtraceGenerator, timerange } from '@kbn/apm-synthtrace-client';
+import { compact, uniq } from 'lodash';
+import { Readable } from 'stream';
+import { ApmSynthtraceEsClient } from '@kbn/apm-synthtrace';
+import type { InternalRequestHeader, RoleCredentials } from '../../../../../shared/services';
+import { APMFtrContextProvider } from '../common/services';
+
+export default function ({ getService }: APMFtrContextProvider) {
+  const apmApiClient = getService('apmApiClient');
+  const svlUserManager = getService('svlUserManager');
+  const svlCommonApi = getService('svlCommonApi');
+  const synthtrace = getService('synthtrace');
+
+  const start = new Date('2022-01-01T00:00:00.000Z').getTime();
+  const end = new Date('2022-01-01T00:15:00.000Z').getTime() - 1;
+
+  async function fetchAndBuildCriticalPathTree(
+    synthtraceEsClient: ApmSynthtraceEsClient,
+    options: {
+      fn: () => SynthtraceGenerator<ApmFields>;
+      roleAuthc: RoleCredentials;
+      internalReqHeader: InternalRequestHeader;
+    } & ({ serviceName: string; transactionName: string } | {})
+  ) {
+    const { fn, roleAuthc, internalReqHeader } = options;
+
+    const generator = fn();
+
+    const unserialized = Array.from(generator);
+    const serialized = unserialized.flatMap((event) => event.serialize());
+    const traceIds = compact(uniq(serialized.map((event) => event['trace.id'])));
+
+    await synthtraceEsClient.index(Readable.from(unserialized));
+
+    return apmApiClient.slsUser({
+      endpoint: 'POST /internal/apm/traces/aggregated_critical_path',
+      params: {
+        body: {
+          start: new Date(start).toISOString(),
+          end: new Date(end).toISOString(),
+          traceIds,
+          serviceName: 'serviceName' in options ? options.serviceName : null,
```

Comment:
Ni: I think we can simplify this
```suggestion
          serviceName: options?.serviceName ?? null,
```
wdyt?

Reply:
`option` is a union type. `options?.serviceName` wouldn't work because typescript can't infer which one of the types in the union it should use.